app/models/factory_log.rb

```diff
@@ -16,29 +18,31 @@ def self.time_ranking
     ranking.sort_by!{|l| l[:total_time]}.reverse
   end
 
-  def self.average_time(factory)
-    sum_time(factory) / log_set_with_trait(factory).size
+  #Return specified factory's average exectuaion time.
+  def self.average_time(factory_name_and_traits)
+    logs = FactoryLog.log_set_with_trait(factory_name_and_traits)
+    logs.map{ |log| log.time }.inject{ |sum, n| sum + n } / logs.size
   end
 
-  def self.sum_time(factory)
-    log_set_with_trait(factory).map{|f| f.time}.inject{|fl1, fl2| fl1 + fl2 }
+  #Return specified factory's total exectuaion time.
+  def self.sum_time(factory_name_and_traits)
+    log_set_with_trait(factory_name_and_traits).map{|f| f.time}.inject{|fl1, fl2| fl1 + fl2 }
   end
 
+  #Return all fatcory_name + trait_name convinaitons in your factoy_logs table.
   def self.candidates
     naming_candidates = []
-    FactoryLog.all.each{|log| naming_candidates.push(log.name + log.traits) unless naming_candidates.any?{|cd| cd ==(log.name + log.traits)}}
+    FactoryLog.all.each{|log| naming_candidates.push(log.name + log.traits) unless naming_candidates.any?{ |cd| cd ==(log.name + log.traits) } }
```

Comment:
Line is too long. [144/80]<br>Surrounding space missing for operator '=='.<br>Space missing to the left of {.<br>Space between { and | missing.